features/daemon.feature

```diff
@@ -0,0 +1,228 @@
+Feature: Pro Upgrade Daemon only runs in environments where necessary
+
+    @series.lts
+    @uses.config.contract_token
+    @uses.config.machine_type.gcp.generic
+    Scenario Outline: daemon should run when appropriate on gcp generic lts
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        # verify its enabled, but stops itself when not configured to poll
+        # TODO-BEGIN: remove this chunk once cloud-init supports this feature
+        When I run `touch /run/cloud-init/cloud-id-gce` with sudo
+        When I run `truncate -s 0 /var/log/ubuntu-advantage-daemon.log` with sudo
+        When I run `systemctl restart ubuntu-advantage.service` with sudo
+        # TODO-END: remove this chunk once cloud-init supports this feature
+        When I wait `1` seconds
+        When I run `cat /var/log/ubuntu-advantage-daemon.log` with sudo
+        Then stdout matches regexp:
+        """
+        daemon starting
+        """
+        Then stdout matches regexp:
+        """
+        Configured to not poll for pro license, shutting down
+        """
+        Then stdout matches regexp:
+        """
+        daemon ending
+        """
+        # verify it stays on when configured to do so
+        When I create the file `/etc/ubuntu-advantage/uaclient.conf` with the following:
+        """
+        ua_config:
+            poll_for_pro_license: true
```

Comment:
Why do we need this extra config here for the daemon to run ? Wouldn't having the cloud-init gce file be enough for it ?